config/config.py

```diff
@@ -87,32 +181,39 @@ def read_config_option(key, expected_type=None, default_value=None):
     """
     logging.debug("Reading config option {} with expected type {}".format(key, expected_type))
 
-    global conf_parser
+    global _conf_parser
     try:
         if not expected_type:
-            value = conf_parser.get("Settings", key)
+            value = _conf_parser.get("Settings", key)
             logging.debug("Got configuration for key {}: {}".format(key, value))
-            return conf_parser.get("Settings", key)
+            return _conf_parser.get("Settings", key)
         elif expected_type is bool:
-            return conf_parser.getboolean("Settings", key)
-    except (ValueError, NoOptionError) as e:
+            return _conf_parser.getboolean("Settings", key)
+    except (ValueError, NoOptionError):
         if default_value:
             return default_value
         else:
             raise
 
 
-def write_config_option(field_name, field_value):
+def _update_config_option(field_name, field_value):
     """
-    Writes an option to the configuration file.
-
+    Sets an option in the config parser (Does not write to file)
     :param field_name: Field to change the value of
-    :param field_value: Value to write to file
-    :return: None
+    :param field_value: Value to change to
+    :return:
     """
-    global conf_parser
-    global user_config_file
-    conf_parser.set("Settings", field_name, field_value)
+    global _conf_parser
+    _conf_parser.set("Settings", field_name, field_value)
+
 
-    with open(user_config_file, 'w') as c_file:
-        conf_parser.write(c_file)
+def write_config_options_to_file():
```

Comment:
Does this get used except just to write the defaults?

Reply:
Not in the CLI uploader, but in my WIP gui uploader, this gets used when you set your config settings via the gui.

Reply:
Ok thanks.